### DISCUSSION TITLE
Switch to meson

### DISCUSSION THREAD

**Discussion by emfend** [viveris/uMTP-Responder#79] [Ideas]
I would like to add a function to the uMTP responder that will notify systemd when the endpoint setup is complete and the gadget can be activated. This would allow the use of standard systemd services that ensure the correct execution order, and unstable workarounds like "sleep 1" used in the demo script can be avoided.
Since this feature should be optional and introduces a dependency on libsystem, a build-time option should be used.

Since there are already some options and it would actually make sense to control the different device speeds via a build configuration as well, it would make sense to switch the build system to meson.

Before I start moving to Meson I would like to know if this is desired at all, or if there is a way to include it in the main branch of uMTP-Responder.

**Comment by emfend** [comment]
I created a [branch ](https://github.com/emfend/uMTP-Responder/tree/meson-notify-fixes) to demonstrate what the project could look like using Meson as the build system.

Additionally, I added an optional systemd notification function. On the one hand, this is a useful feature for avoiding the ugly time-based workaround to delay gadget activation, and on the other hand, it shows how easy it is to add options with additional dependencies.

**Reply by mriesch-wv** [reply]
Nice work! Maybe it would help the discussion if you opened a pull request.

**Comment by jfdelnero** [comment]
Hello,

We can add the meson support but for the moment we need to keep the makefile support too, this is required by some of our internal projects.

Thanks for the systemd support, i will merge the modifications. 
Many embedded linux don't have the systemd layer. i have to find an alternate sync method for these cases.
